File: Hash Table/3005. Count Elements With Maximum Frequency.cpp
class Solution {
    public:
        int maxFrequencyElements(vector<int>& nums) {
            unordered_map<int, int> frequency;  // Map to store frequency of each number
            
            // Count occurrences of each number in the array
            for (int num : nums) {
                frequency[num]++;
            }
            
            int maxFreq = 0;  // Variable to track the highest frequency
            for (const auto& pair : frequency) {
                maxFreq = max(maxFreq, pair.second);  // Find the maximum frequency
            }
            
            int sum = 0;  // Sum of occurrences of elements with max frequency
            for (const auto& pair : frequency) {
                if (pair.second == maxFreq) {
                    sum += pair.second;
                }
            }
            
            return sum;
        }
    };
    

File: Hash Table/3005. Count Elements With Maximum Frequency.java
class Solution {
    public int maxFrequencyElements(int[] nums) {
        // Map to store the frequency of each number
        Map<Integer, Integer> frequencyMap = new HashMap<>();

        int maxFrequency = 0;   // Highest frequency found so far
        int maxFrequencyCount = 0; // Number of elements that have the highest frequency

        // Count the frequency of each number
        for (int num : nums) {
            frequencyMap.put(num, frequencyMap.getOrDefault(num, 0) + 1);
        }

        // Find the maximum frequency and how many numbers have it
        for (int frequency : frequencyMap.values()) {
            if (frequency > maxFrequency) {
                maxFrequency = frequency;   // Update max frequency
                maxFrequencyCount = 0;      // Reset count of elements with max frequency
            }
            if (frequency == maxFrequency) {
                maxFrequencyCount++;        // Count how many numbers have the max frequency
            }
        }

        // Each of these numbers contributes 'maxFrequency' times
        return maxFrequencyCount * maxFrequency;
    }
}
